src/services/refactors/convertArrowFunctionOrFunctionExpression.ts

```diff
@@ -0,0 +1,185 @@
+                }
+                else {
+                    body = func.body;
+                }
+
+                newNode = createArrowFunction(func.modifiers, func.typeParameters, func.parameters, func.type, /* equalsGreaterThanToken */ undefined, body);
+                edits = textChanges.ChangeTracker.with(context, t => t.replaceNode(file, func, newNode));
+                break;
+
+            default:
+                Debug.fail("invalid action");
+                break;
+        }
+
+        return { renameFilename: undefined, renameLocation: undefined, edits };
+    }
+
+    function getInfo(file: SourceFile, startPosition: number): BasicInfo | undefined {
+        const token = getTokenAtPosition(file, startPosition);
+        let maybeFunc;
+
+        maybeFunc = getArrowFunctionFromDeclaration(token.parent);
+        if (!!maybeFunc) return { fromDeclaration: true, func: maybeFunc };
+
+        maybeFunc = getContainingFunction(token);
+        if (!!maybeFunc && (isFunctionExpression(maybeFunc) || isArrowFunction(maybeFunc)) && !rangeContainsRange(maybeFunc.body, token)) {
+                return { fromDeclaration: false, func: maybeFunc };
+        }
+
+        return undefined;
+    }
+
+    function getArrowFunctionFromDeclaration(parent: Node): ArrowFunction | undefined {
+        if (!(isVariableDeclaration(parent) || (isVariableDeclarationList(parent) && parent.declarations.length === 1))) return undefined;
+        const variableDeclaration = isVariableDeclaration(parent) ? parent : parent.declarations[0];
+
+        if (!variableDeclaration.initializer || !isArrowFunction(variableDeclaration.initializer)) return undefined;
+        return variableDeclaration.initializer;
+    }
+
+    function makeBlock(func: ArrowFunction | FunctionExpression): Block {
+        if (isExpression(func.body)) {
+            const statements: Statement[] = [createReturn(func.body)];
+            return createBlock(statements, /* multiLine */ true);
+        }
+        else {
+            return func.body;
+        }
+    }
+
+    function makeFunctionDeclaration(func: FunctionExpression | ArrowFunction, statement: VariableStatement, name: Identifier, body: Block) {
+        return createFunctionDeclaration(
+            func.decorators,
+            statement.modifiers,
+            func.asteriskToken,
+            name,
+            func.typeParameters,
+            func.parameters,
+            func.type,
+            body);
+    }
+
+    function getHeaderInfo(func: FunctionExpression | ArrowFunction): HeaderInfo | undefined {
+        const variableDeclaration = func.parent;
+        if (!isVariableDeclaration(variableDeclaration) || !isVariableDeclarationInVariableStatement(variableDeclaration)) return undefined;
+
+        const variableDeclarationList = findAncestor(variableDeclaration, n => isVariableDeclarationList(n))!;

```

Comment:
Changes:
```suggestion
        const variableDeclarationList = variableDeclaration.parent;
```
Reasoning:
- simpler and more readable
- no double assertion